relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -411,29 +413,104 @@ fn extract_span_metrics(
+    let mut shared_tags = BTreeMap::new();
+
+    if let Some(environment) = event.environment.as_str() {
+        shared_tags.insert("environment".to_owned(), environment.to_owned());
+    }
+
+    if let Some(transaction_name) = event.transaction.value() {
+        shared_tags.insert("transaction".to_owned(), transaction_name.to_owned());
+    }
 
-        if let Some(transaction_name) = event.transaction.value() {
-            shared_tags.insert("transaction".to_owned(), transaction_name.to_owned());
+    if let Some(trace_context) = get_trace_context(event) {
+        if let Some(op) = extract_transaction_op(trace_context) {
+            shared_tags.insert("transaction.op".to_owned(), op);
         }
-        // TODO(iker): must add shared tags: environment, op, release, etc
-
-        if let Some(user) = event.user.value() {
-            if let Some(value) = get_eventuser_tag(user) {
-                metrics.push(Metric::new_mri(
-                    MetricNamespace::Spans,
-                    "user",
-                    MetricUnit::None,
-                    MetricValue::set_from_str(&value),
-                    timestamp,
-                    shared_tags.clone(),
-                ));
+    }
+
+    for annotated_span in spans {
+        if let Some(span) = annotated_span.value() {
+            let mut span_tags = shared_tags.clone();
+
+            if let Some(span_op) = span.op.value() {
+                span_tags.insert("op".to_owned(), span_op.to_owned());
+
+                let span_module = if span_op.starts_with("http") {
+                    Some("http")
+                } else if span_op.starts_with("db") {
+                    Some("db")
+                } else if span_op.starts_with("cache") {
+                    Some("cache")
+                } else {
+                    None
+                };
+
+                if let Some(module) = span_module {
+                    span_tags.insert("module".to_owned(), module.to_owned());
+                }
+
+                let action = match span_module {
+                    Some("http") => span
+                        .data
+                        .value()
+                        // TODO(iker): some SDKs extract this as method
+                        .and_then(|v| v.get("http.method"))
+                        .and_then(|db_op| db_op.as_str()),
+                    Some("db") => span
+                        .data
+                        .value()
+                        .and_then(|v| v.get("db.operation"))
+                        .and_then(|db_op| db_op.as_str()),
+                    _ => None,
+                };
+
+                if let Some(act) = action {
+                    span_tags.insert("action".to_owned(), act.to_owned());
+                }
+
+                // TBD product requirements, adding it as a placeholder
+                span_tags.insert("domain".to_owned(), "todo".to_owned());
```

Comment:
Does it even make sense to set the tag if we do not know the value yet?

Reply:
I discussed this with Jan, and he suggested adding it as a placeholder. I don't think it really helps, so I removed it; we could apply the same reasoning for the other unimplemented tags I assume.